NGINX/nginx/ingest/parser.yml

```
@@ -0,0 +1,141 @@
+name: nginx
+pipeline:
+  - name: parsed_event
+    external:
+      name: grok.match
+      properties:
+        pattern: '%{ERROR}|%{COMBINED}'
+        custom_patterns:
+          NGINXURL: '(%{URIPROTO:url_scheme}://)?(?:%{USER:url_username}(?::[^@]*)?@)?(?:%{URIHOST:url_domain})?(?:%{URIPATHPARAM:url_path})'
+          ERROR: '%{YEAR}/%{MONTHNUM}/%{MONTHDAY} %{HOUR}:%{MINUTE}:%{SECOND} \[%{WORD}\] %{NUMBER}#%{NUMBER}: (\*%{NUMBER} )?(%{DATA}client:\s+%{IP:source_address})?(%{DATA}server:\s+%{IPORHOST:destination_address})?(%{DATA}request: \"%{WORD:http_request_method} %{NGINXURL:url_original} HTTP/%{NUMBER:http_version}\")?%{GREEDYDATA}'
+          COMBINED: '(%{IPORHOST:destination_address} )?%{IP:source_address} - %{DATA:source_user_name} \[%{HTTPDATE}\] \"%{WORD:http_request_method} %{NOTSPACE:url_original} HTTP/%{NUMBER:http_version}\" %{NUMBER:http_response_status_code} (-|%{NUMBER:http_response_bytes})? \"%{DATA:http_request_referrer}\" \"%{DATA:user_agent_original}\"( \"%{DATA}\")?( \"%{DATA}\")?( \"%{DATA}\")?( \"%{DATA:http_response_mimetype}\")?'
+    filter: '{{original.message | re_match("\{(?:[^{}]|())*\}") == False}}'
+  - name: parsed_event
+    external:
+      name: json.parse-json
+    filter: '{{original.message | re_match("\{(?:[^{}]|())*\}")}}'
+  - name: set_json_fields
+    filter: '{{original.message | re_match("\{(?:[^{}]|())*\}")}}'
+  - name: set_grok_fields
+    filter: '{{original.message | re_match("\{(?:[^{}]|())*\}") == False}}'
+
+stages:
+  set_json_fields:
```

Suggesting added this constant ECS fields (maybe `event.type` should be adapted according to the original event: is an access log or an error?)
```suggestion
   set_ecs_fields:
    actions:
      - set:
          event.kind: 'event'
          event.category: ['web']
          event.type: ['access']
          event.dataset: 'access'
          observer.product: 'nginx'
          observer.type: 'WEB server'
          observer.vendor: 'F5'
    set_json_fields:
```